xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -649,20 +661,28 @@ BuildObject(CFileItem&                    item,
         }
         else if (!subtitles.empty())
         {
-            /* trying to find subtitle with prefered language settings */
-            std::string preferredLanguage = (CServiceBroker::GetSettingsComponent()->GetSettings()->GetSetting("locale.subtitlelanguage"))->ToString();
-            std::string preferredLanguageCode;
-            g_LangCodeExpander.ConvertToISO6392B(preferredLanguage, preferredLanguageCode);
+          std::string preferredLanguage{"en"};
 
-            for (unsigned int i = 0; i < subtitles.size(); i++)
-            {
-                ExternalStreamInfo info = CUtil::GetExternalStreamDetailsFromFilename(file_path.GetChars(), subtitles[i]);
+          /* trying to find subtitle with prefered language settings */
+          auto setting = settings->GetSetting("locale.subtitlelanguage");
```

Comment:
Is there a potential concern about the lifetime of settings pointer object in a multi threaded life?
Theres a few for loops scattered through some of the code paths up to this point. Do we need to worry that settings may get blown away by another thread (technically only on shutdown i guess really?) before getting called here?

Reply:
I don't think so really. Settings is such a fundamental part that if it ever went away on use we _should_ fail.

Reply:
all good then